src/EditorFeatures/Core/ExternalAccess/IntelliCode/IntentProcessor.cs

```diff
@@ -23,12 +26,19 @@ namespace Microsoft.CodeAnalysis.ExternalAccess.IntelliCode
     internal class IntentSourceProvider : IIntentSourceProvider
     {
         private readonly ImmutableDictionary<(string LanguageName, string IntentName), Lazy<IIntentProvider, IIntentProviderMetadata>> _lazyIntentProviders;
+        private readonly JsonSerializerOptions _serializerOptions;
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public IntentSourceProvider([ImportMany] IEnumerable<Lazy<IIntentProvider, IIntentProviderMetadata>> lazyIntentProviders)
         {
             _lazyIntentProviders = CreateProviderMap(lazyIntentProviders);
+
+            _serializerOptions = new JsonSerializerOptions
+            {
+                PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+            };
+            _serializerOptions.Converters.Add(new JsonStringEnumConverter(JsonNamingPolicy.CamelCase));
```

Comment:
one concern i have is how are json exceptions handled when you use this?  i think we need to presume the data may not be well formed, and we should be resilient to that.

Reply:
Done - added a report and catch to the deserialization path so that we know when it goes wrong.  Since intellicode is providing the json, we definitely want to be notified if it is malformed.